ControlSettings/ControlSettingsCommon.h

```diff
@@ -0,0 +1,78 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef CONTROLSETTINGS_COMMON_H
+#define CONTROLSETTINGS_COMMON_H
+
+
+#define VIDEO_DESCRIPTION_MAX (25)
+#define VIDEO_DESCRIPTION_NAME_SIZE (25)
+
+#define DECLARE_JSON_RPC_METHOD(method) \
+uint32_t method(const JsonObject& parameters, JsonObject& response);
+
+#define PLUGIN_Lock(lock) pthread_mutex_lock(&lock)
+#define PLUGIN_Unlock(lock) pthread_mutex_unlock(&lock)
+
+#define returnResponse(return_status, error_log) \
```

Comment:
please do not use params "success" or "error"-s

this is legacy of old project XRE Service Manager

we are not using it any more

the only thing which is required from you is return error code

e.g.
Core::ERROR_NONE
Core::ERROR_BAD_REQUEST
...

and document errors in the JSON RPC interface schema

Reply:
Thanks for your comments.
Just pushed initial changes. Will remove returnResponse( ) and just return the Core::ERROR_NONE (or) Core::ERROR_BAD_REQUEST.